scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamFunctions.scala

```diff
@@ -58,4 +62,20 @@ object BeamFunctions {
       }
     }
   }
+
+  case class HashJoinFn[K, V, U, W](
+    joiner: (K, V, Iterable[U]) => Iterator[W],
+    sideInput: PCollectionView[java.util.Map[K, java.lang.Iterable[U]]]
+  ) extends DoFn[(K, V), (K, W)] {
+    @ProcessElement
+    def processElement(c: DoFn[(K, V), (K, W)]#ProcessContext): Unit = {
+      val sideInputMap = c.sideInput(sideInput)
```

Comment:
are we not materializing this more than one? Should we not do something like:
```scala
private[this] var mapRight: java.util.Map[K, java.lang.Iterable[U]] = null

    @ProcessElement
    def processElement(c: DoFn[(K, V), (K, W)]#ProcessContext): Unit = {
      if (mapRight == null) {
        mapRight = c.sideInput(sideInput)
      }

      ...

```
Then in FinishBundle set it back to `null`.